Cubical/HITs/Susp/SuspProduct.agda

```diff
@@ -0,0 +1,232 @@
+
+  A□2≃ : A□2 span ≃ typ A
+  A□2≃ = invEquiv (pushoutIdfunEquiv' _)
+
+  f : typ A → B ⋁ A
+  f = inr
+
+  g : typ A → C ⋁ A
+  g = inr
+
+  A□○≃ : A□○ span ≃ Pushout f g
+  A□○≃ = pushoutEquiv _ _ _ _
+    A□2≃ (idEquiv _) (idEquiv _)
+    (funExt λ
+      { (inl x) → push _
+      ; (inr x) → refl
+      ; (push a i) j → sq (~ i) (~ j) })
+    (funExt λ
+      { (inl x) → push _
+      ; (inr x) → refl
+      ; (push a i) j → sq (~ i) (~ j) })
+    where
+      sq : ∀ {ℓ} {C : Type ℓ} {c : C}
+        → Square refl (sym (Pushout.push _))
+          refl (sym (push {f = λ x → c} {g = λ _ → pt A}
+            tt ∙ refl))
+      sq = slideSquare (sym (rUnit _))
+
+  A4□≃ : A4□ span ≃ typ A
+  A4□≃ = invEquiv (pushoutIdfunEquiv _)
+
+  A2□≃ : A2□ span ≃ Unit
+  A2□≃ = invEquiv (pushoutIdfunEquiv _)
+
+  A2□isContr : isContr (A2□ span)
+  A2□isContr = isOfHLevelRespectEquiv 0 (pushoutIdfunEquiv _) isContrUnit
+
+  A○□≃ : A○□ span ≃ (B ⋁∙ₗ C) ⋁ A
+  A○□≃ = pushoutEquiv _ _ _ _
+    A2□≃ (idEquiv _) A4□≃
+    (lemma A2□isContr refl) (lemma A2□isContr refl)
+    where
+      -- functions on contractible domains are determined by the basepoint
+      lemma : ∀ {ℓ ℓ'}
+        → {A : Type ℓ} {B : Type ℓ'}
+        → {f g : A → B}
+        → (c : isContr A)
+        → f (c .fst) ≡ g (c .fst)
+        → f ≡ g
+      lemma {f = f} {g} c p = funExt λ x
+        → cong f (sym (c .snd x))
+        ∙ p
+        ∙ cong g (c .snd x)
+
+  wedgePushout : PushoutSquare
+  wedgePushout = extendPushoutSquare
+    (pushoutToSquare record { f1 = f ; f3 = g })
+    (invEquiv A□○≃ ∙ₑ pathToEquiv (3x3-lemma span) ∙ₑ A○□≃)
+
+
+module _ {ℓ ℓ'} (X : Pointed ℓ) (Y : Pointed ℓ') where
+  open commSquare
+  open 3-span
+
+  {-
+    X × Y ---> Y --------> 1
+      ↓   A    ↓     C     ↓
+      X ---> Σ(X∧Y) --> ΣY ∨ Σ(X∧Y)
+      ↓   B    ↓     D     ↓
+      1 -> ΣX ∨ Σ(X∧Y) --> ? = ΣX ∨ ΣY ∨ Σ(X∧Y)
+  -}
+
+  pushoutA₀ : PushoutSquare  -- inrP maps to south, but we want north
+  pushoutA₀ = extendPushoutSquare
+      (pushoutToSquare record { f1 = fst ; f3 = snd })
+      (pathToEquiv (joinPushout≡join _ _) ∙ₑ
+        invEquiv (isoToEquiv (SmashJoinIso {A = X} {B = Y})))
+
+  squareA : I → commSquare
+  squareA i = record
+    { commSquare (pushoutA₀ .fst)
+    ; inrP = λ _ → merid (inl _) i
+    ; comm = transp (λ j →
+      Path (typ X × typ Y → Susp (X ⋀ Y))
+        (λ _ → north) (λ _ → merid (inl _) (i ∨ ~ j)))
+      i (pushoutA₀ .fst .commSquare.comm)
+    }
+
+  pushoutA : PushoutSquare
+  pushoutA = squareA i0 ,
+    subst isPushoutSquare (λ i → squareA (~ i)) (pushoutA₀ .snd)
+
+  module _ where
+    open PushoutNull (typ X) (Susp (X ⋀ Y)) north
+    open PushoutPasteDown pushoutA
+      (squareL .sp .f1)
+      (squareL .inlP)
+      (squareL .inrP)
+      (squareL .commSquare.comm)
+
+    pushoutB : PushoutSquare
+    pushoutB = pushoutSquareL
+
+    pushoutAB : PushoutSquare
+    pushoutAB = _ , isPushoutBottomSquare→isPushoutTotSquare (pushoutB .snd)
```

Comment:
(replying to @aljungstrom on another issue) It's this line and/or possibly other similar lines. Agda is checking the middle function of the two squares agree, so that we can apply the pasting lemma. Strangely when I evaluated each field of the commSquare record they returned instantly, but it gets stuck for a couple seconds and a few GB of memory when I try to evaluate the entire record.

Reply:
See the comments slightly above for a commutative diagram of what's happening.